### PR TITLE
Release v0.34.22

### PR DESCRIPTION
**Pull request by thanethomson**
:book: [CHANGELOG](https://github.com/tendermint/tendermint/blob/release/v0.34.22/CHANGELOG.md#v03422)

---

#### PR checklist

- [x] Tests written/updated, or no tests needed
- [x] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [x] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,37 @@
...
+  fixed
+- [p2p] [\#9500](https://github.com/tendermint/tendermint/issues/9500) Prevent
+  peers who have errored being added to the peer set (@jmalicevic)
+- [indexer] [\#9473](https://github.com/tendermint/tendermint/issues/9473) Fix
+  bug that caused the psql indexer to index empty blocks whenever one of the
+  transactions returned a non zero code. The relevant deduplication logic has
+  been moved within the kv indexer only (@cmwaters)
+- [blocksync] [\#9518](https://github.com/tendermint/tendermint/issues/9518) A
+  block sync stall was observed during our QA process whereby the node was
+  unable to make progress for a long period and then eventually recovered.
```
Comment (williambanfield) [inline CHANGELOG.md:33]: I didn't see it eventually recover, I observed it stall indefinitely for hours until the node was killed.

**Review by cmwaters — APPROVED** [review]
This looks all fine to me

File: CHANGELOG.md (hunk 1 above)
Comment (thanethomson) [reply CHANGELOG.md]: Updated. How's the new wording?

**Review by sergio-mena — APPROVED** [review]

**Review by williambanfield — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (williambanfield) [reply CHANGELOG.md]: Looks great!

**Pull request merged**